### PR TITLE
Set LRU values to zero to prevent cleanup when using encryption

### PR DESCRIPTION
**Pull request by Half-Shot**
Requires https://github.com/vector-im/matrix-bot-sdk/pull/6

Should fix https://github.com/matrix-org/matrix-hookshot/issues/609, although other process crashing bugs may remain.

### CODE REVIEW

File: src/appservice.ts [hunk 1]
```diff
@@ -38,6 +38,10 @@ export function getAppservice(config: BridgeConfig, registration: IAppserviceReg
         },
         storage: storage,
         intentOptions: {
+            // If encryption support is enabled, we cannot expire Intent objects or we risk
+            // a resource contention on the Sled DB.
+            maxCached: config.encryption && 0,
+            maxAgeMs: config.encryption && 0,
```
Comment (jaller94) [inline src/appservice.ts:44]: Better use something like `config.encryption ? foo : 0` to limit what falsey values are passed to the Intent.
```
> true && 0
0
> false && 0
false
> null && 0
null
> undefined && 0
undefined
```

**Review by jaller94 — APPROVED** [review]
Yes, but please limit possible types.

File: src/appservice.ts (hunk 1 above)
Comment (Half-Shot) [reply src/appservice.ts:44]: Don't we trust `config.encryption` to be object|undefined though, as that's strongly typed in the config.

**Comment by Half-Shot** [comment]
We're not going to do this, in favour of replacing Sled.

**Pull request closed without merging**